scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -4736,7 +4736,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
         Template(Nil, Nil, self, body)
       }
     } else {
-      val parents = templateParents(afterExtend)
+      val parents = if (token.is[Colon]) Nil else templateParents(afterExtend)
```

Comment:
should this whole function be changed to:
```
    val edefsOrTemplate = if (token.is[LeftBrace]) {
      // @S: pre template body cannot stub like post body can!
      val (self, body) = templateBody(enumCaseAllowed)
      if (token.is[KwWith] && self.name.is[Name.Anonymous] && self.decltpe.isEmpty) {
        val edefs = body.map(ensureEarlyDef)
        next()
        Left(edefs)
      } else {
        Right(Template(Nil, Nil, self, body))
      }
    } else {
      Left(Nil)
    }
    edefsOrTemplate match {
      case Right(t) => t
      case Left(edefs) =>
        val parents = if (token.is[Colon]) Nil else templateParents(afterExtend)
        template(edefs, parents, enumCaseAllowed)
    }
```

Reply:
We should be fine as is, since early defs are not allowed in Dotty, so they will not exist with colons.

It would be probably good to remove early defs from Scala 3 dialect, but we could do that separately if needed.